06_puppy/hsy3418/types.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+// Puppy defines the data structure corresponding to a pet
+type Puppy struct {
+	ID     int32   `json:"id"`
+	Value  float32 `json:"value"`
+	Breed  string  `json:"breed"`
+	Colour string  `json:"colour"`
+}
+
+//Storer define standard CRUD operations for puppys
+type Storer interface {
+	CreatePuppy(Puppy) error
```

Comment:
IMO Create should provision a system unique ID and return it - the user/client of this API probably doesn't have complete knowledge of all other entries. Happy for you to argue against it (leave it).

Reply:
resolving the comment is not exactly arguing against it ;) .
please put forward your case why you think the user/client of your system would have knowledge of (available) ID or change implementation to provision on ID (e.g. by incrementing a counter).

Reply:
I agree with @juliaogris on this one. The user should not be expected to understand the complete set of entries. All they want to do is create a puppy, and then have some method of retrieving that puppy. Having the ID returned means the API handles that logic and the user doesn't need to worry about what else exists. They have a means of accessing their object through the id returned from a successful create. 

@hsy3418 if you don't agree or have some other view / ideas, please speak up and let us know. I would be happy to discuss and hear your thoughts. Alternatively, please update this one accordingly.

Reply:
@juliaogris @rokane  I totally agree with you the client who is going to create the puppy wouldn't know the ID, they will just enter the puppy info and ID should be handled in the back-end logic, didn't think about that deeply before.

Reply:
By returning the ID as a sign for creating the puppy successfully make sense to the client definitely, I think I will change the method accordingly, but my question is if there is no error return, would it work similarly to the client? Cause my understanding is if the client just wants to know if the puppy been created or not, they don't really need the ID, unless the front-end logic needs to show the exact ID. Correct me if I am wrong.

Reply:
I am going to take that back :), just rethink about it, since the ID is created in backend, there is no error could be returned since there wouldn't be duplicate puppys.